lib/msf/core/db_manager/http/servlet/host_servlet.rb

```diff
@@ -8,11 +8,16 @@ def self.api_path_with_id
     "#{HostServlet.api_path}/?:id?"
   end
 
+  def self.api_search_path
+    "#{HostServlet.api_path}/search"
+  end
+
   def self.registered(app)
     app.get HostServlet.api_path_with_id, &get_host
     app.post HostServlet.api_path, &report_host
     app.put HostServlet.api_path_with_id, &update_host
     app.delete HostServlet.api_path, &delete_host
+    app.post HostServlet.api_search_path, &search
```

Comment:
Why is this needed when the `HostServlet.get_host` method and by proxy the `DBManager::Host.hosts` method already provide a means for finding hosts?

Reply:
I think this is cleanup I missed

Reply:
I only need to cleanup the "puts" - this is actually needed to satisfy findorcreate,  what was there is wrong, it should be findhosts - there is actually a mapping DBManager::Host.get_host for retrieving a single host

Reply:
actually I need to fix this